integration/spark-common/src/main/scala/org/apache/carbondata/spark/util/GlobalDictionaryUtil.scala

```diff
@@ -784,11 +784,24 @@ object GlobalDictionaryUtil {
       }
     } catch {
       case ex: Exception =>
-        LOGGER.error(ex, "generate global dictionary failed")
-        throw ex
+        ex match {
+          case spx: SparkException =>
+            LOGGER.error(spx, "generate global dictionary failed")
+            throw new Exception(
+              trimErrorMessage(spx.getMessage))
+          case _ =>
+            LOGGER.error(ex, "generate global dictionary failed")
+            throw ex
+        }
     }
   }
 
+  // Get proper error message of TextParsingException
+  def trimErrorMessage(input: String): String = {
+    val message = input.split("Hint")(0).split("TextParsingException: ")(1)
```

Comment:
There may be chances of input is null because `spsx.getMessage()` can be null, so handle NPE here.
And also please check the array length after you split with `"Hint"`, other wise we may get ArrayIndexOutOfBoundException.